src/Data/Symbol.purs

```diff
@@ -8,11 +8,19 @@ module Data.Symbol
 import Type.Proxy (Proxy(..))
 
 -- | A value-level proxy for a type-level symbol.
+-- | **Deprecated as of v0.14.0 PureScript release**: use `Type.Proxy` instead.
 data SProxy (sym :: Symbol) = SProxy
 
 -- | A class for known symbols
 class IsSymbol (sym :: Symbol) where
-  reflectSymbol :: Proxy sym -> String
+  -- Note: Before v0.14.0, we did not have polykinds. Thus, we needed
+  -- kind-specific proxy types to pass a Symbol around.
+  -- Once v0.14.0 was released, we could use the kind-generic `Proxy` type
+  -- insteand. However, to reduce the code breakage, we're using
+  -- `forall proxy. proxy sym` here so that `SProxy` code will still compile.
+  -- When PureScript makes a new breaking release after the v0.14.0 release,
+  -- this type signature will be updated to `Proxy sym -> String`.
```

Comment:
We should probably make an issue to track this so we don't forget.

Reply:
Open it up!

Reply:
@hdgarrood or @natefaubion, do you have suggestions on what that issue ought to touch on?

Reply:
Nope, this seems fairly self-explanatory to me. We can always discuss further on the issue too.